addons/account/models/account_tax.py

```diff
@@ -1240,7 +1240,7 @@ def _eval_taxes_computation(self, taxes_computation, evaluation_context):
                     tax_amount = 0.0
                 tax_data['tax_amount'] = tax_amount
                 tax_data['tax_amount_factorized'] = tax_data['tax_amount'] * tax_data['_factor']
-                if rounding_method == 'round_per_line' or (not special_mode and tax_data['price_include'] and round_price_include):
+                if rounding_method == 'round_per_line' or ((special_mode == 'total_included' or tax_data['price_include']) and round_price_include):
```

Comment:
You need to keep the old condition about special_mode but add your new one:
```suggestion
                if rounding_method == 'round_per_line' or ((special_mode == 'total_included' or (not special_mode and tax_data['price_include'])) and round_price_include):
```
(Reorder the condition as you like to make it readable plz)